### DISCUSSION TITLE
为什么部署好docker 后，网站访问不了？

### DISCUSSION THREAD

**Discussion by a506488043** [0xJacky/nginx-ui#709] [Q&A]
Docker 部署好，网站添加，刚开始能正常访问。后面就都不能访问了？这是什么情况？
偏好设置，监听主机是空的，跟这个有关系吗？怎么解决？
![image](https://github.com/user-attachments/assets/cebf2a1b-2dee-4471-b1e8-7f34794e370e)

**Comment by 0xJacky** [comment]
请提供日志，谢谢

**Comment by a506488043** [comment]
怎么提供日志？日志的文件路径是nginx的吗

**Comment by 0xJacky** [comment]
请提供以下信息：
1. 宿主机操作系统，版本
2. Docker/Docker Compose 启动命令
3. docker logs 容器名获取容器日志

**Comment by a506488043** [comment]
> 请提供以下信息：
> 
> 1. 宿主机操作系统，版本
> 2. Docker/Docker Compose 启动命令
> 3. docker logs 容器名获取容器日志
这个是日志

[nginx_ui.log](https://github.com/user-attachments/files/17690042/nginx_ui.log)

**Comment by 0xJacky** [comment]
nginx ui，可以正常访问？但是自己的站点不能访问吗

**Comment by a506488043** [comment]
> 怎么提供日志？日志的文件路径是nginx的吗

怎么时好时坏的？
![image](https://github.com/user-attachments/assets/1894b16e-75ec-4257-be67-8e5df348625c)

**Comment by 0xJacky** [comment]
这个建议检查你的网络配置，或者查看 docker 的 network 是否发生冲突

**Comment by a506488043** [comment]
> nginx ui，可以正常访问？但是自己的站点不能访问吗

对的，nginx ui 能访问，自己的站点访问不了。现在又能访问了；时好时坏的

**Comment by 0xJacky** [comment]
自己的站点，配置信息也贴一下

**Comment by a506488043** [comment]
> 自己的站点，配置信息也贴一下

![image](https://github.com/user-attachments/assets/2b8c3296-f7e7-4fad-a667-bb4bb7246bbf)
![image](https://github.com/user-attachments/assets/877631c4-dfc7-4a02-9421-bfc701a40800)

**Comment by a506488043** [comment]
> 自己的站点，配置信息也贴一下

nginx ui 能访问，自己的站点不应该访问不了；

**Comment by 0xJacky** [comment]
打开高级编辑模式，将配置文件全部贴过来

**Comment by a506488043** [comment]
> 打开高级编辑模式，将配置文件全部贴过来
127.0.0.1 不是docker 里面的IP，是Ubuntu 的IP 

![image](https://github.com/user-attachments/assets/d85a0192-d31d-43aa-8271-f86407982da3)

**Comment by 0xJacky** [comment]
这个不对吧，为什么这个配置文件看不到 docker registry repo 的后端

**Comment by a506488043** [comment]
> 这个不对吧，为什么这个配置文件看不到 docker registry repo 的后端
docker registry repo 的后端 在哪里看？
![image](https://github.com/user-attachments/assets/977c5ce6-f18f-4b66-9747-ed7924d86442)

**Comment by 0xJacky** [comment]
嗯？你的 Ubuntu 上面是不是还装了个 Nginx，还有，提供一下 docker 启动 nginx-ui 的命令，如果是 compose 启动的，提供 docker-compose.yml。

**Comment by a506488043** [comment]
> 嗯？你的 Ubuntu 上面是不是还装了个 Nginx，还有，提供一下 docker 启动 nginx-ui 的命令，如果是 compose 启动的，提供 docker-compose.yml。

Ubuntu 已经卸载了；
docker run -dit  --name=nginx-ui  --restart=always  -e TZ=Asia/Shanghai  -v /mnt/256G/docker/nginx_ui/nginx:/etc/nginx  -v /mnt/256G/docker/nginx_ui/nginx-ui:/etc/nginx-ui  -v /var/www:/var/www  -p 80:80 -p 443:443   uozi/nginx-ui:latest

**Comment by 0xJacky** [comment]
那你还有其他配置文件吗？就这一个配置文件？

**Comment by 0xJacky** [comment]
应该先排查一下看看是哪个配置文件给 docker registry repo 做了反向代理

**Comment by a506488043** [comment]
> docker registry repo 是指 docker.saiita.cn的配置吗？

**Comment by a506488043** [comment]
> 应该先排查一下看看是哪个配置文件给 docker registry repo 做了反向代理

奇怪了，电脑能访问，手机访问不了。

**Comment by 0xJacky** [comment]
你现在不是要访问这个 UI 吗
<img width="148" alt="image" src="https://github.com/user-attachments/assets/cbe13143-cc3b-483a-a9a4-96904ae3fe68">

这个 UI 是怎么部署的，部署在哪个机器

**Comment by a506488043** [comment]
> 你现在不是要访问这个 UI 吗 <img alt="image" width="148" src="https://private-user-images.githubusercontent.com/13096985/384656666-cbe13143-cc3b-483a-a9a4-96904ae3fe68.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyMTAyODIsIm5iZiI6MTczMTIwOTk4MiwicGF0aCI6Ii8xMzA5Njk4NS8zODQ2NTY2NjYtY2JlMTMxNDMtY2MzYi00ODNhLWE5YTQtOTY5MDRhZTNmZTY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEwVDAzMzk0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzZDUwYWJlYTdkNWNhODNhNDE0YzkzMTZlM2YyOTY1ZjNkYWQwNDVkZTNmN2QxYTQ0MTJkYTc1MWI3YzQyNTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YR7GTZiZ1hCWLUeU14X9e0L4qKsCKw7xEZx_jeRO_N0">
> 
> 这个 UI 是怎么部署的，部署在哪个机器

UI 部署在家里的ITX 上的。手机和电脑都在局域网上；我用 docker Nginx manager 是没问题的；Nginx manager 已经卸载了。
用路由器 指定域名 到了ITX上的IP了。

**Comment by a506488043** [comment]
> 你现在不是要访问这个 UI 吗 <img alt="image" width="148" src="https://private-user-images.githubusercontent.com/13096985/384656666-cbe13143-cc3b-483a-a9a4-96904ae3fe68.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyMTM2NTgsIm5iZiI6MTczMTIxMzM1OCwicGF0aCI6Ii8xMzA5Njk4NS8zODQ2NTY2NjYtY2JlMTMxNDMtY2MzYi00ODNhLWE5YTQtOTY5MDRhZTNmZTY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEwVDA0MzU1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEzMGZkMzIyN2I4MGM4N2FjZDY2NDhhZjIxOGE1ODRiODk1YTFjODc2NDJhMjg2ODdlNjFjOTEyZmEzMGY0ZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PH6nHkGFYT0JKDJqNUxIvUycNVy-tSz40Vss8Pn2HNw">
> 
> 这个 UI 是怎么部署的，部署在哪个机器

有毒，手机PING网站也是局域网，手机访问不了；
电脑能访问；

**Comment by 0xJacky** [comment]
做了虚拟组网？

**Reply by a506488043** [reply]
> 做了虚拟组网？

没有呀，就是路由器的局域网

**Reply by a506488043** [reply]
> > 做了虚拟组网？
> 
> 没有呀，就是路由器的局域网

用脚本部署，也是不行；能连接上docker.saiita.cn，是因为开了VPN，连接到外网了，本地有FRP到外网的；
就是不知道为什么，脚本安装和docker安装，配置的网站打不开？

**Comment by a506488043** [comment]
总感觉你们demo里的nginx配置有问题。
